### DISCUSSION TITLE
iterate calendar date

### DISCUSSION THREAD

**Discussion by twilson9034** [Budibase/budibase#3455] [Help]
I would like to be able to setup recurring events without having to input each date.

Any ideas on how I could accomplish this?

**Comment by mjashanks** [comment]
If you used an external database - such as SQL, you could combine this with Javascript transformers. You could
1. Save some data in a row, that represented something like, for example `Day: Monday, Repitition: Weekly`
2. Add a custom query to `select` this row from the datasource
3. Add a transformer that understands `Day: Monday, Repitition: Weekly`, and returns a collection dates that apply.

There's going to have to be some custom code somewhere in there, unless we add some native concept of "Recurring Event".

One other way that we **might in future** achieve this, is by using an automation to generate new dates, on a CRON (scheduled) trigger.

It would look like
1. Query template date e.g. `Day: Monday, Repitition: Weekly`
2. Add a Javascript block to determine all these future recurring events
3. Use a `Foreach` block (we don't have this, but would like to), to save every generated date as a row.

I'm adding this 2nd method as a reminder to ourselves of the need for this `foreach` block!

**Comment by twilson9034** [comment]
I'm strictly using bb DB. Is this possible within budibase?

**Reply by mjashanks** [reply]
I don't think so at the moment :(

**Comment by twilson9034** [comment]
Is there anyway to change the Date selector in automation to be able to accept code, then we could use foreach right?

**Reply by mjashanks** [reply]
I don't think this will work, there is no way to trigger multiple parallel actions in an automation. I.e. foreach.

**Reply by twilson9034** [reply]
If I wanted to have all the dates in the same row where each date would be comma separated.
